core/google/cloud/iam.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Non-API-specific IAM policy definitions
+
+For allowed roles / permissions, see:
+https://cloud.google.com/iam/docs/understanding-roles
+"""
+
+# Generic IAM roles
+
+OWNER_ROLE = 'roles/owner'
+"""Generic role implying all rights to an object."""
+
+EDITOR_ROLE = 'roles/editor'
+"""Generic role implying rights to modify an object."""
+
+VIEWER_ROLE = 'roles/viewer'
+"""Generic role implying rights to access an object."""
+
+
+class Policy(object):
+    """IAM Policy
+
+    See:
+    https://cloud.google.com/iam/reference/rest/v1/Policy
+
+    :type etag: str
+    :param etag: ETag used to identify a unique of the policy
+
+    :type version: int
+    :param version: unique version of the policy
+    """
+    _OWNER_ROLES = (OWNER_ROLE,)
+    """Roles mapped onto our ``owners`` attribute."""
+
+    _EDITOR_ROLES = (EDITOR_ROLE,)
+    """Roles mapped onto our ``editors`` attribute."""
+
+    _VIEWER_ROLES = (VIEWER_ROLE,)
+    """Roles mapped onto our ``viewers`` attribute."""
+
+    def __init__(self, etag=None, version=None):
+        self.etag = etag
+        self.version = version
+        self.bindings = {}
+
+    @property
+    def owners(self):
+        """Legacy access to owner role."""
+        result = set()
+        for role in self._OWNER_ROLES:
+            for member in self.bindings.get(role, ()):
+                result.add(member)
+        return result
```

Comment:
Hmm I don't think that's what I was suggesting =/

* base `iam.Policy` object has a `owners` attribute which *only* tracks `roles/owner`.  This can be readwrite. It turns out that `roles/owner` *is* actually grantable on pubsub resources. We just recommend against it so maybe we an add a warning, but it won't break AFAICT
* `pubsub.iam.Policy` has an `admins` attribute which *only* tracks `roles/pubsub.admin`, warning on `policy.owners` suggests converting calls to `policy.owner` to calls to `policy.admins`. This is also readwrite

Under this solution, what the library returns for a given call, and does for a given call will change, but all calls will still work. The change is just accurately reflecting the API instead.

Alternatively we could do as @tseaver and simply silently convert `policy.owners` to refer exclusively to `roles/pubsub.admin`, and remove all ability to address legacy roles, which I'm also fine with.